Fix #1760: Have a dedicated URL for iframed explorations, and make the embedding script minimal

MAKOSCAFEE opened the pull request.

seanlip commented on core/templates/dev/head/player/conversation_skin_embed_directive.html:
```diff
@@ -0,0 +1,800 @@
+        <div class="col-1-4">
+          <div class="logo-bottom-nav-bar">
+            <div class="nav-bar-bottom-visit-oppia">
+              <span ng-if="isOnTerminalCard()" class="oppia-more-exploratotion">For more explorations, visit:</span>
```
Replicating comment from previous PR: Class is misspelled. Should be something like oppia-view-more-explorations

MAKOSCAFEE replied:
done

seanlip commented:
Replicating notes from previous version of PR:

(1) The number box is too far to the left. Why is this? (Should also check that the other interactions display correctly.)

![scr-3](https://cloud.githubusercontent.com/assets/10575562/17275127/2f50d320-56b2-11e6-8186-966fa6130695.png)

(2) In the following screenshot:
- the "Click on the Map" button doesn't look like a button, although it does in the usual conversation view. Can this be fixed? Otherwise, it's not clear what to do. 
- try clicking the correct spot on the map with the window having smallish height. You'll notice that the "Hooray, you found it!" card is now behind the bottom green bar, making further progress impossible. Could this card's position adjust to be a bit higher, but only if the window height is too small?

![scr-4](https://cloud.githubusercontent.com/assets/10575562/17275130/3e6f1588-56b2-11e6-91bd-fd62fce2d517.png)

(3) "Restart exploration" on small screens is too small to read comfortably. Can that font be a bit bigger? If there's no space, truncate to "Restart" or get rid of it entirely.

(4) In the image below, you'll see that the text on the right of the progress dots isn't quite vertically-centered. Could this be fixed?

![note-5](https://cloud.githubusercontent.com/assets/10575562/17358217/9f15c732-5915-11e6-97dc-bf24210594d9.png)

(5) What are your plans for the new scripts file, as referenced at the end of my previous comment? Is that something you'd like to fix now? If so, let's do that. If not, then we still need to make sure that the `?iframed=true` thing works -- it's currently broken. I lean towards the former option (fixing it with a redirect, and updating the embedding instructions in the editor), but I'd like to know what you think, too.

seanlip commented:
Hi @MAKOSCAFEE -- FYI, I've done (5) as we discussed. There are two more things that need to be done as a follow up:

- [ ] Make the initial load of the iframe faster
- [ ] Have the exploration embed handler use a suitable template when there's a 404 error (instead of the default 404 template)

If you like, I can try taking a look into both of these. Also, let me know if you have any questions/concerns about my previous commit.

Thanks!

MAKOSCAFEE commented:
Hi @seanlip  I have addressed all the concerns from the previous PR. can you take a look?. Also if you can look at the issues you mentioned above it will be great.
Thanks.

seanlip commented:
Sorry, it looks like there are still problems.

(1) The number box alignment is still wrong. It should be like this (the left edge of the input box should align with the left edge of the card):

![want](https://cloud.githubusercontent.com/assets/10575562/18080610/1c636fce-6e4b-11e6-91a2-8855542dba98.png)

but it currently looks like this:

![now](https://cloud.githubusercontent.com/assets/10575562/18080615/211fa58c-6e4b-11e6-975c-654e50e223f6.png)

(2) If the page content is larger than the height, it is possible to scroll below the footer. This looks really bad (to reproduce, make the page height small and navigate to the state with the map):

![2424-2](https://cloud.githubusercontent.com/assets/10575562/18080637/377b53c6-6e4b-11e6-9853-82ba9b081e84.png)

Instead, the footer should always remain fixed to the bottom of the page, and scrolling should happen _behind_ the footer.

(3) In the following screenshot, the Oppia text on the bottom right is a bit too stretched out. Can we give it a max width so that it doesn't get distorted?

![2424-4](https://cloud.githubusercontent.com/assets/10575562/18080685/5ddc7b8a-6e4b-11e6-8b57-5ac0ebb01cb9.png)

Thanks!

seanlip commented on core/templates/dev/head/player/conversation_skin_embed_directive.html:
```diff
@@ -0,0 +1,808 @@
+          </progress-dots>
+        </div>
+        <div class="col-1-4">
+          <div class="logo-bottom-nav-bar">
```
Why are you using the same class for this div as well as the img tag five lines below?

seanlip commented:
Hi @MAKOSCAFEE, I took a look at the embedding e2e test errors, and I'm pretty sure they're due to your modifications. Did you try replicating the test in core/tests/protractor/embedding.js on your local machine? You can also read through the test in that file itself, and follow along manually -- this should give you a feel for what's going wrong.

seanlip commented:
Ah, ignore the last comment. I did exactly that, and found that the version number was not being passed along correctly. I tracked this down to a redirection error in the reader.py controller file and will shortly be pushing a commit to fix it.

seanlip commented:
Hi @MAKOSCAFEE, I think I've done all my parts. Please fix the remaining issues, and then I'll take a final clean up pass. Thanks!

seanlip commented:
@MAKOSCAFEE -- ping?

seanlip commented:
Hi @MAKOSCAFEE -- is everything fixed and ready for review, or do you still have more to do?

Also, could you please merge develop into this branch?

Thanks!

MAKOSCAFEE commented:
Hi @seanlip I think all the 3 above comments were addressed. you can do a review.

Thanks.

MAKOSCAFEE commented:
ooh and I am merging the develop branch now.

seanlip commented on core/templates/dev/head/player/conversation_skin_embed_directive.html:
```diff
@@ -140,7 +141,7 @@ <h4 class="conversation-skin-responses-dropdown-container" ng-click="toggleShowP
         </div>
         <div class="col-1-4">
-          <div class="logo-bottom-nav-bar">
+          <div class="bottom-nav-bar">
```
Where is bottom-nav-bar defined? Should we just get rid of this class?

MAKOSCAFEE replied:
removed

seanlip commented on main.py:
```diff
@@ -235,6 +235,9 @@ def ui_access_wrapper(self, *args, **kwargs):
         reader.ExplorationPage),
     get_redirect_route(
+        r'%s/<exploration_id>' % feconf.EXPLORATION_URL_EMBED_PREFIX,
+        reader.ExplorationPageEmbed, 'exploration_embed_page'),
```
@MAKOSCAFEE did you run the devserver after you merged? This line breaks it...

MAKOSCAFEE replied:
Fixed

seanlip commented:
Hi @MAKOSCAFEE -- I'm having a huge amount of trouble testing this; everything is erroring. Did you run the devserver after the merge for a sanity check? If not, could you please do so and fix the issues so that I can make follow-up commits on this?

Thanks!

MAKOSCAFEE commented:
Hi @seanlip  I have fixed the break that was introduced in the last commit...two test continue to fail though. redirect in embedding.js I guess.

seanlip commented:
Hi @MAKOSCAFEE -- I noticed you push a commit that seems to cause backend tests to fail (in addition to the e2e test which is still failing). Was that intentional?
